src/components/messaging-provider.js

```diff
@@ -98,6 +94,13 @@ class MessagingProvider extends Component {
       web3Account
     } = this.props
 
```

Comment:
Nitpick: this will work well since `MessagingProvider` never un-mounts when users navigate to different pages. From purely puristic view if this component were ever to be used differently where it un-mounts each initialisation of the component would trigger another interval and not clean it up when it un-mounts. One way to fix this: 
```
componentWillUnmount() {
    if (this.notificationsInterval) {
      clearInterval(this.notificationsInterval)
    }
  }
```

Reply:
😳 @sparrowDom I thought about adding that, but didn't think it would be necessary. It's fair to point out that others may not recognize the need for it in the future.